dace/transformation/dataflow/hbm_copy_transform.py

```diff
@@ -0,0 +1,163 @@
+        allow_none=False,
+        desc="If set storage types will be set to CPU Heap if it is Default for the two involved arrays")
+
+    def _get_split_size(self, virtual_shape: Iterable,
+                        split_count: List[int]) -> List[int]:
+        """
+        :return: the shape of a part-array on one HBMbank
+        """
+        new_shape_list = []
+        for d in range(len(virtual_shape)):
+            if split_count[d] != 1:
+                new_shape_list.append(virtual_shape[d] // split_count[d])
+            else:
+                new_shape_list.append(virtual_shape[d])
+        return new_shape_list
+
+    @staticmethod
+    def can_be_applied(graph: Union[SDFG, SDFGState],
+                       candidate: Dict['PatternNode', int], expr_index: int,
+                       sdfg: SDFG, strict: bool) -> bool:
+        src = graph.nodes()[candidate[HbmCopyTransform._src_node]]
+        dst = graph.nodes()[candidate[HbmCopyTransform._dst_node]]
+        src_array = sdfg.arrays[src.data]
+        dst_array = sdfg.arrays[dst.data]
+
+        # same dimensions means HBM-array needs 1 dimension more
+        collect_src = len(src_array.shape) - 1 == len(dst_array.shape)
+        distribute_dst = len(src_array.shape) + 1 == len(dst_array.shape)
+        if collect_src:
+            try:
+                tmp = int(src_array.shape[0])
+            except:
+                return False
+        elif distribute_dst:
+            try:
+                tmp = int(dst_array.shape[0])
+            except:
+                return False
+        return collect_src or distribute_dst
+
+    @staticmethod
+    def expressions():
+        return [
+            utils.node_path_graph(HbmCopyTransform._src_node,
+                                  HbmCopyTransform._dst_node)
+        ]
+
+    def apply(self, sdfg: SDFG) -> Union[Any, None]:
+        # Load/parse infos from the SDFG
+        graph = sdfg.nodes()[self.state_id]
+        src = graph.nodes()[self.subgraph[HbmCopyTransform._src_node]]
+        dst = graph.nodes()[self.subgraph[HbmCopyTransform._dst_node]]
+        src_array = sdfg.arrays[src.data]
+        dst_array = sdfg.arrays[dst.data]
+        collect_src = len(src_array.shape) - 1 == len(
+            dst_array.shape
+        )  # If this is not true we have to distribute to dst (checked in can_apply)
+        if collect_src:
+            bank_count = int(src_array.shape[0])
+            true_size = dst_array.shape
+        else:
+            bank_count = int(dst_array.shape[0])
+            true_size = src_array.shape
+        ndim = len(true_size)
+
+        # Move Default to CPU_Heap
+        if self.default_to_cpu_storage:
+            if sdfg.arrays[src.data].storage == dtypes.StorageType.Default:
+                sdfg.arrays[src.data].storage = dtypes.StorageType.CPU_Heap
+            if sdfg.arrays[dst.data].storage == dtypes.StorageType.Default:
+                sdfg.arrays[dst.data].storage = dtypes.StorageType.CPU_Heap
```

Comment:
Unfortunatelly not, because the AccessNodes are placed inside of Unrolled Maps, and they don't have a default storage type. (And I'm hesitant to just change it to CPU_Heap in general)

Reply:
So the codegen would currently throw an error? Maybe the property should at least let the user choose what the default is, instead of always using `CPU_Heap`.

Reply:
Yes it throws key errors. Hm ok, I can change it to just accept a StorageType